src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialsStore.java

```diff
@@ -0,0 +1,135 @@
+package com.cloudbees.jenkins.plugins.kubernetes_credentials_provider;
+
+import java.util.List;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.acegisecurity.Authentication;
+import org.apache.commons.lang.NotImplementedException;
+import org.jenkins.ui.icon.Icon;
+import org.jenkins.ui.icon.IconSet;
+import org.jenkins.ui.icon.IconType;
+import org.kohsuke.stapler.export.ExportedBean;
+import hudson.model.ModelObject;
+import hudson.security.ACL;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.CredentialsStoreAction;
+import com.cloudbees.plugins.credentials.domains.Domain;
+
+public class KubernetesCredentialsStore extends CredentialsStore {
+
+    private final KubernetesCredentialProvider provider;
+    private final KubernetesCredentialsStoreAction action = new KubernetesCredentialsStoreAction(this);
+
+    public KubernetesCredentialsStore(KubernetesCredentialProvider provider) {
+        super(KubernetesCredentialProvider.class);
+        this.provider = provider;
+    }
+
+    @NonNull
+    @Override
+    public ModelObject getContext() {
+        return Jenkins.getInstance();
+    }
+
+    @Override
+    public boolean hasPermission(@NonNull Authentication authentication, @NonNull Permission permission) {
+        return Jenkins.getInstance().getACL().hasPermission(authentication, permission);
+    }
+
+    @NonNull
+    @Override
+    public List<Credentials> getCredentials(@NonNull Domain domain) {
+        // TODO: Filter by domain - how do I do this?
+        return  provider.getCredentials(Credentials.class, Jenkins.getInstance(), ACL.SYSTEM);
+    }
+
+    @Override
+    public boolean addCredentials(@NonNull Domain domain, @NonNull Credentials credentials) {
+        throw new NotImplementedException();
```

Comment:
should only throw an IOException (as per parent signature)

Reply:
`NotImplementedException` (and your suggestion of `UnsupportedOperationException` for that matter) are both `RuntimeException`s